lms/validation/_api.py

```diff
@@ -69,3 +70,36 @@ class APIRecordResultSchema(JSONPyramidRequestSchema):
     """
     Score — i.e. grade — for this submission. A value between 0 and 1, inclusive.
     """
+
+
+class APICreateAssignmentSchema(PyramidRequestSchema):
+    """Schema for validating assignment creation requests made by our frontend."""
+
+    class Content(Schema):
+        class File(Schema):
+            display_name = fields.Str(required=True)
+            id = fields.Int(required=True)
+            updated_at = fields.Str(required=True)
+            size = fields.Int(required=True)
+
+        type = fields.Str(
+            required=True, validate=marshmallow.validate.OneOf(["url", "file"])
+        )
+        url = fields.Url(required=False, allow_none=True)
+        file = fields.Nested(File, required=False, allow_none=True)
+
+    ext_lti_assignment_id = fields.Str(required=True)
+    """Canvas only assignment unique identifier"""
+
+    course_id = fields.Str(required=True)
+    """Course ID for the assignment. We'd need it to construct the document_url"""
+
+    groupset = fields.Integer(required=False, allow_none=True)
+    """Groupset the assignment belongs to if created as a smalls groups assignment"""
+
+    content = fields.Nested(Content, required=True)
+
+    @validates_schema
+    def validate(self, data, **_kwargs):
+        if data["content"]["type"] == "url" and not data["content"].get("url"):
+            raise ValidationError("url is mandatory with type is 'url'")
```

Comment:
And `data["content"]["file"]` is mandatory if `data["content"]["type"] == "file"` right?